kernel/src/process_manager/process.rs

```diff
@@ -171,9 +173,14 @@ impl TrustedProcess {
 
     }
 
-    pub fn trustlet(parent: ProcessID, _data: u64, _size: u64, _pgt: u64) -> Self{
+    pub fn trustlet(parent: ProcessID, data: u64, size: u64, pgt: u64) -> Self{
         // Inherit the data from the Zygote
         let trustlet = TrustedProcess::dublicate(parent);
+        if data != 0 {
+            let (function_code, function_code_range) = ProcessPageTableRef::copy_data_from_guest(data, size, pgt);
+            trustlet.base.page_table_ref.add_function(function_code, size);
```

Comment:
`page_table_ref.add_function()` (more precicely, `add_region_vaddr()` expects page-aligned size, therefore
```suggestion
            let size = (4096 - (size & 0xFFF)) + size;
            trustlet.base.page_table_ref.add_function(function_code, size);
```

(I think `add_region_vaddr` should be able to take non-page aligend size though)